enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+
+#### CRI-O Changes
+
+CRI-O will be updated to support a new configuration value that a user
+can specify, `mgmt_ctr_cpuset`.
+
+```ini
+[crio.runtime]
+mgmt_ctr_cpuset = “0-1”
+```
+
+This field describes the CPU set that management workloads will be
+configured to use.
+
+CRI-O will be configured to support a new annotation on pods,
+`io.openshift.management.cores`.
+
+```ini
+[crio.runtime.runtimes.runc]
+  runtime_path = "/usr/bin/runc"
+  runtime_type = "oci"
+  allowed_annotations = ["io.openshift.management.cores"]
+```
+
+Pods that have the `io.openshift.management.cores` annotation will
+have their cpuset configured to the value in `mgmt_ctr_cpuset`, as
+well as have their CPU shares configured to the value of the
+annotation.
+
+Note that this field does not conflict with the `infra_ctr_cpuset`
+config option, as the infra container will still be put in that
+cpuset.  They can be configured as the same value if the infra
+container should also be considered to be managed.
+
+#### API Server Admission Hook
+
+A new admission hook in the kubernetes API server within OpenShift
+will mutate pods when they are created to make 2 changes.
+
+1. It will move CPU requests to "management core" requests, so that
+   the scheduler can successfully place the pod on the node, even
+   though the sum of the CPU requests for all management workloads may
+   exceed the actual CPU capacity of the management CPU pool.
+2. It will add an annotation, `io.openshift.management.cores` with a
+   value equal to the original CPU requests, so that CRI-O can use the
+   value to configure the CPU shares for the container.
+
+We will not change pods in a way that changes their quality-of-service
+class. So, we would not strip CPU requests unless they also have
+memory requests, because if we mutate the pod so that it has no CPU or
+memory requests the quality-of-service class of the pod would be
+changed automatically. Any labeled pod that is already BestEffort
+would be annotated using `0` as the value so that CRI-O will have an
+indicator to configure the CPU shares as BestEffort.
+
+#### Kubelet Changes
+
+Kubelet will be changed to look for a configuration file,
+`/etc/kubernetes/management-pinning`, to enable the management
+workload partitioning feature. The file should contain the cpuset
+specifier for the CPUs making up the management CPU pool.
+
+Kubelet will be changed so that when the feature is enabled, [when a
+static pod definition is
+read](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/file_linux.go)
+it is mutated in a way similar to the API server admission hook will
+mutate regular pods.
+
+Kubelet will be changed so that when the feature is enabled, it
+advertises a new extended resource of "management cores", representing
+all of the CPU capacity of the host (not just the management CPU
+pool).
+
+#### Installer Changes
+
+The installer will be changed to accept the new `managementCPUIDs`
```

Comment:
How are the `managementCPUIDs` predicted? I saw a reference to integer counts earlier; is that the only component of the ID (e.g. some logical integer count for the available CPUs?) Would the same set of logical IDs (e.g. `1-2`) be appropriate for 10 different clusters configured via the same `install-config.yaml`?

Reply:
the enumerated ids may vary if hyperthreading is on/off , and if system is multi or single numa.  customer would know where they want us isolated.